pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -151,4 +152,29 @@ public static double[] arrayConcatDouble(double[] values1, double[] values2) {
   public static String[] arrayConcatString(String[] values1, String[] values2) {
     return ArrayUtils.addAll(values1, values2);
   }
+
+  @ScalarFunction
+  public static int arrayElementAtInt(int[] arr, int idx) {
+    return idx > 0 && idx <= arr.length ? arr[idx - 1] : (int) DataSchema.ColumnDataType.INT.getNullPlaceholder();
```

Comment:
@walterddr Pinot Strings are zero-based while Presto's are one-based. Does it make sense to support one-based arrays in Pinot and zero-based strings?
cc: @Jackie-Jiang

Reply:
Seems Postgres is also one-based. We might want to change the string functions to one-based as well. Currently `substr` is zero-based. @nizarhejazi Do you see other zero-based string functions?

Reply:
Hmm, seems there are other zero-based string functions such as `strpos`, `split_part` etc.

Reply:
also there's no such function as `substr` in postgres. only `substring` and it is one-based.

Reply:
`substr` is supported in presto, but is one-based: https://prestodb.io/docs/current/functions/string.html

Reply:
Fix split not on whole separator #9650